randobot/handler.py

```diff
@@ -146,13 +165,190 @@ async def break_notifications(self):
                 sleep((interval - duration - datetime.timedelta(minutes=5)).total_seconds()),
             )
 
+    async def ex_draft(self, args, message):
+        """
+        Handle !draft commands.
+
+        Set up room for Draft Mode.
+        """
+        if self._race_in_progress() or not self._is_tournament_match():
+            return
+        elif self._is_tournament_match and self.data.get('entrants_count') < 2:
+            await self.send_message(
+                'Both runners must be present before enabling Draft Mode.'
+            )
+            return
+        elif self._is_tournament_match and self.data.get('entrants_count') > 2:
+            await self.send_message(
+                'Draft Mode is only available for head-to-head matches.'
+            )
+            return
+        if len(args) == 1 and args[0] in ('on', 'off'):
+            if args[0] == 'on' and not self.state['draft_mode']:
+                self.state['draft_mode'] = True
+                await gather(
+                    self.send_message(
+                        'Welcome to OoTR Draft Mode! '
+                        'You can disable Draft Mode at any time with !draft off.'
+                    ),
+                    self.ex_fpa(args, message)
+                )
+                
+                entrants = await self._is_qualified()
+                # If we can't verify qualification data, disable Draft Mode
+                if len(entrants) < 2:
+                    await gather(
+                        self.send_message(
+                            'Error fetching runner data. Please contact a tournament organizer.'
+                        ),
+                        self.ex_draft(['off'], message)
+                    )
+                
+                
+                await self.send_message(
+                    f"{entrants[0]['name']}, please select whether or not to ban first with !first or !second."
+                )
+                self.zsr.draft_data.update({
```

Comment:
`self.zsr` is shared between all race handlers, which means this would get overwritten if there were multiple draft races going on at the same time. This data should live in `self.state` instead (which also preserves it in case the handler disconnects)